File: csharp/src/code-smells/bloaters/LongParameterList.cs
namespace RefactoringAvanzado.CodeSmells.Bloaters;

public class ReportGenerator
{
    public void GenerateReport(
        string title,
        DateTime startDate,
        DateTime endDate,
        bool includeCharts,
        bool includeSummary,
        string authorName,
        string authorEmail)
    {
        Console.WriteLine($"Generando reporte: {title}");
        Console.WriteLine($"Desde {startDate.ToShortDateString()} hasta {endDate.ToShortDateString()}");
        Console.WriteLine($"Autor: {authorName} ({authorEmail})");
        if (includeCharts) Console.WriteLine("Incluyendo gráficos...");
        if (includeSummary) Console.WriteLine("Incluyendo resumen...");
        Console.WriteLine("Reporte generado exitosamente.");
    }
}

public static class LongParameterListDemo
{
    public static void DemoLongParameterList()
    {
        var gen = new ReportGenerator();
        gen.GenerateReport(
            "Ventas Q1",
            new DateTime(2025, 1, 1),
            new DateTime(2025, 3, 31),
            true,
            false,
            "Pat Smith",
            "pat@example.com"
        );
    }
}

File: typescript/src/code-smells/bloaters/long-parameter-list.ts
class ReportGenerator {
  generateReport(
    title: string,
    startDate: Date,
    endDate: Date,
    includeCharts: boolean,
    includeSummary: boolean,
    authorName: string,
    authorEmail: string,
  ) {
    console.log(`Generando reporte: ${title}`)
    console.log(`Desde ${startDate.toDateString()} hasta ${endDate.toDateString()}`)
    console.log(`Autor: ${authorName} (${authorEmail})`)
    if (includeCharts) console.log('Incluyendo gráficos...')
    if (includeSummary) console.log('Incluyendo resumen...')
    console.log('Reporte generado exitosamente.')
  }
}

export function demoLongParameterList(): void {
  const gen = new ReportGenerator()
  gen.generateReport(
    'Ventas Q1',
    new Date('2025-01-01'),
    new Date('2025-03-31'),
    true,
    false,
    'Pat Smith',
    'pat@example.com',
  )
}
